### PR TITLE
feat : add nb-psc-mig-l7xlb sample and tests

### PR DESCRIPTION
**Pull request by tanmaybangale**
What's changed, or what was fixed?

- sample for l7xlb->psc->mig->apigee 
- tests for the sample

**Fixes:** #add

- [X] I have run all the tests locally and they all pass.
- [X] I have followed the relevant style guide for my changes.

### CODE REVIEW

File: samples/x-nb-psc-mig-l7xlb/README.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# Apigee X exposed in Multiple GCP Regions with PSC and MIG External HTTPS Load Balancer
+
+End-to-end sample terraform code to provision Apigee X exposed by External Load Balancer with PSC and MIG.
+
+If you plan to use PSC for Apigee northbound network routing, follow the instructions in this document to configure active health check. 
+At this time, PSC does not support active health check monitoring. To work around this limitation of PSC, you can modify the Apigee installation configuration to use a managed instance group (MIG), which does provide active health check capability. Refer the network diagram below
```
Comment (danistrebel) [inline samples/x-nb-psc-mig-l7xlb/README.md:6]: Minor detail but in my understanding the missing health check is a problem of PSC-NEGs not the PSC. Can we maybe also link to this [doc](https://cloud.google.com/load-balancing/docs/negs) and add a link to this solution from the NB PSC with Neg Example?

File: samples/x-nb-psc-mig-l7xlb/x-demo.tfvars [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  "testgroup" = { hostnames = ["example.xyz.com"] }
+  # Add more environment groups if needed
+}
+
+apigee_instances = {
+  "euw1-instance" = {
+    region       = "europe-west1"    
+    ip_range     = "10.1.4.0/22"              
+    environments = ["test"] 
+  },
```
Comment (danistrebel) [inline samples/x-nb-psc-mig-l7xlb/x-demo.tfvars:43]: Remove?

File: samples/x-nb-psc-mig-l7xlb/x-demo.tfvars [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+exposure_subnets = [
+  {
+    name               = "apigee-exposure-1"
+    ip_cidr_range      = "10.100.0.0/24"
+    region             = "europe-west1"
+    instance           = "euw1-instance"
+    secondary_ip_range = null
+  },
+  /*
```
Comment (danistrebel) [inline samples/x-nb-psc-mig-l7xlb/x-demo.tfvars:73]: remove?

File: samples/x-nb-psc-mig-l7xlb/x-demo.tfvars [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+psc_subnets = [
+    {
+    name               = "psc-subnet-1"
+    ip_cidr_range      = "10.100.255.240/29"
+    region             = "europe-west1"
+    instance           = "euw1-instance"
+    secondary_ip_range = null
+  },
+  /*
```
Comment (danistrebel) [inline samples/x-nb-psc-mig-l7xlb/x-demo.tfvars:92]: remove?

**Review by danistrebel — APPROVED** [review]
Please see minor comments. Rest looks great! thanks so much for creating this PR!

**Comment by danistrebel** [comment]
regarding the automated checks we can ignore the conventional commits lint. We just use them to automatically generate the release notes (e.g. a commit message of `feat: foo` creates a new minor version with an entry foo in the release notes).

For the docs there seems to be some inconsistency in `samples/x-nb-psc-mig-l7xlb/README.md` can you make sure the doc is automatically generated with the script in the tools folder?

File: samples/x-nb-psc-mig-l7xlb/x-demo.tfvars (hunk 3 above)
Comment (tanmaybangale) [reply samples/x-nb-psc-mig-l7xlb/x-demo.tfvars]: removed it

**Comment by danistrebel** [comment]
Great work @tanmaybangale ! thanks again!

**Pull request merged**